rocketpy/motors/TankGeometry.py

```diff
@@ -164,6 +169,91 @@ def inverse_volume(self):
             lambda v: v / (np.pi * self.average_radius**2),
         )
 
+    @cache
+    def balance(self, lower, upper):
+        """
+        The volume balance of the tank as a function of height.
+
+        Returns
+        -------
+        Function
+            Tank centroid as a function of height.
+        """
+        height = self.area.identityFunction()
+
+        # Tolerance of 1e-8 is used to avoid numerical errors
+        upper = upper + 1e-12 if upper - lower < 1e-8 else upper
+
+        balance = (height * self.area).integralFunction(lower, upper)
+
+        # Correct naming
+        balance.setInputs("height (m)")
+        balance.setOutputs("balance (m⁴)")
+
+        return balance
+
+    @cache
+    def Ix_volume(self, lower, upper):
+        """
+        The volume of inertia of the tank with respect to
+        the x-axis as a function of height. The x direction is
+        assumed to be perpendicular to the motor body axis.
+
+        Returns
+        -------
+        Function
+            Tank volume of inertia as a function of height.
+        """
+        height2 = self.radius.identityFunction() ** 2
+
+        # Tolerance of 1e-8 is used to avoid numerical errors
+        upper = upper + 1e-12 if upper - lower < 1e-8 else upper
+
+        inertia = (self.area * (height2 + self.radius**2 / 4)).integralFunction(
+            lower, upper
+        )
+
+        # Correct naming
+        inertia.setInputs("height (m)")
+        inertia.setOutputs("volume of inertia (m⁵)")
+
+        return inertia
+
+    @cache
+    def Iy_volume(self, lower, upper):
+        """
+        The volume of inertia of the tank with respect to
+        the y-axis as a function of height. The y direction is
+        assumed to be perpendicular to the motor body axis.
+
+        Due to symmetry, this is the same as the Ix_volume.
+
+        Returns
+        -------
+        Function
+            Tank volume of inertia as a function of height.
+        """
+        return self.Ix_volume(lower, upper)
+
+    @cache
+    def Iz_volume(self, lower, upper):
+        """
+        The volume of inertia of the tank with respect to
+        the z-axis as a function of height. The z direction is
+        assumed to be parallel to the motor body axis.
+
+        Returns
+        -------
+        Function
+            Tank volume of inertia as a function of height.
+        """
+        # Tolerance of 1e-8 is used to avoid numerical errors
+        upper = upper + 1e-12 if upper - lower < 1e-8 else upper
+
+        inertia = (self.area * self.radius**2).integralFunction(lower, upper) / 2
```

Comment:
Just noting that this integral is also correct.

Reply:
Thank you for the double check.